arkouda/io.py

```diff
@@ -831,30 +760,560 @@ def export(
+    string.
 
-    prefix, extension = os.path.splitext(read_path)
-    first_file = f"{prefix}_LOCALE0000{extension}"
-    filetype = get_filetype(first_file)
+    If datasets is None, infer the names of datasets from the first file
+    and read all of them. Use ``get_datasets`` to show the names of datasets
+    to HDF5/Parquet files.
 
-    if filetype not in ["HDF5", "Parquet"]:
-        raise RuntimeError(
-            "File type not supported. Import is only supported for HDF5 and Parquet file formats."
+    Examples
+    --------
+    Read with file Extension
+    >>> x = ak.read('path/name_prefix.h5') # load HDF5 - processing determines file type not extension
+    Read without file Extension
+    >>> x = ak.read('path/name_prefix.parquet', file_format='Parquet') # load Parquet
+    Read Glob Expression
+    >>> x = ak.read('path/name_prefix*') # Reads HDF5
+    """
+    warn(
+        "ak.pdarrayIO.read has been deprecated. Please use ak.IO.read_parquet or ak.IO.read_hdf",
+        DeprecationWarning,
+    )
+    if isinstance(filenames, str):
+        filenames = [filenames]
+    datasets = _prep_datasets(filenames, datasets, allow_errors)
+
+    file_format = file_format.lower()
+    if file_format == "infer":
+        cmd = "readany"
+    elif file_format == "hdf5":
+        cmd = "readAllHdf"
+    elif file_format == "parquet":
+        cmd = "readAllParquet"
+    else:
+        warnings.warn(f"Unrecognized file format string: {file_format}. Inferring file type")
+        cmd = "readany"
+    if iterative:  # iterative calls to server readhdf
+        return {
+            dset: read(
+                filenames,
+                dset,
+                strictTypes=strictTypes,
+                allow_errors=allow_errors,
+                iterative=False,
+                calc_string_offsets=calc_string_offsets,
+            )[dset]
+            for dset in datasets
+        }
+    else:
+        rep_msg = generic_msg(
+            cmd=cmd,
+            args={
+                "strict_types": strictTypes,
+                "dset_size": len(datasets),
+                "filename_size": len(filenames),
+                "allow_errors": allow_errors,
+                "calc_string_offsets": calc_string_offsets,
+                "dsets": datasets,
+                "filenames": filenames,
+            },
         )
+        rep = json.loads(rep_msg)  # See GenSymIO._buildReadAllHdfMsgJson for json structure
+        items = rep["items"] if "items" in rep else []
+        file_errors = rep["file_errors"] if "file_errors" in rep else []
+        if allow_errors and file_errors:
+            file_error_count = rep["file_error_count"] if "file_error_count" in rep else -1
+            warnings.warn(
+                f"There were {file_error_count} errors reading files on the server. "
+                + f"Sample error messages {file_errors}",
+                RuntimeWarning,
+            )
```

Comment:
Can we call `_parse_errors` here?

Reply:
We can. I did not update anything in the deprecated functions, but I can if we think it is necessary.